src/utils/transaction.ts

```diff
@@ -38,6 +42,168 @@ import { CROSS_CHAIN_ACTION_STATUS } from '../constants/transactionDispatcherCon
 import { ITransactionBlock } from '../types/transactionBlock';
 import { POLYGON_USDC_CONTRACT_ADDRESS } from '../constants/assetConstants';
 
+const fetchBestRoute = async (
+  sdk: EtherspotSdk,
+  fromChainId: number,
+  toChainId: number,
+  fromAmount: BigNumber,
+  fromTokenAddress: string,
+  toTokenAddress: string,
+  toAddress: string
+) => {
+  try {
+    const createTimestamp = +new Date();
+    const routes = await sdk.getAdvanceRoutesLiFi({
+      fromChainId,
```

Comment:
This try block is way too large - ideally we shouldn't be wrapping entire functions in a try block, as this could turn into testing hell, but rather just individual components that actually need to tested (such as interactions with an sdk or fetch calls)

Better to have more but smaller try blocks then one large one